components/script/dom/htmlimageelement.rs

```diff
@@ -1528,6 +1528,15 @@ impl VirtualMethods for HTMLImageElement {
         self.super_type().unwrap().unbind_from_tree(context);
         let document = document_from_node(self);
         document.unregister_responsive_image(self);
+        if let Some(parent) = self.upcast::<Node>().GetParentElement() {
```

Comment:
I think we'll need to check context.parent here instead. Removing the node from the tree may mean that it has no parent element according to the Node APIs. This will address the " removed from a picture parent element" mutation.